mrrc/pkgs/indexing.py

```diff
@@ -0,0 +1,99 @@
+from jinja2 import Template
+import sys
+import subprocess
+from treelib import Tree, Node
+import uuid
+import os
+from typing import List
+
+
+class IndexHTML(object):
+    # object for holding index html file data
+    def __init__(self, title: str, header: str, items: List[str]):
+        self.title = title
+        self.header = header
+        self.items = items
+
+    def generate_index_file_content(self) -> str:
+        template = Template(get_index_template())
+        return template.render(index=self)
+
+
+def get_index_template() -> str:
+    DEFAULT_INDEX_TEMPLATE = os.path.join(
+        os.environ["HOME"], ".mrrc/template/index.html.j2"
+    )
+    with open(DEFAULT_INDEX_TEMPLATE, encoding="utf-8") as file_:
+        return file_.read()
+
+
+def generate_from_local(repo):
```

Comment:
We should never generate from local, because the local repo files is just parts of the real structure of the whole artifacts. The more real process should be
* Files Uploaded to/Deleted from s3
* Get the changed directories from local
* Scan these changed directories from s3
* Generate the indexes based on the whole structure from s3 for these changed directories.